presto-iceberg/src/main/java/com/facebook/presto/iceberg/InMemoryCatalog.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.BaseMetastoreCatalog;
+import org.apache.iceberg.BaseMetastoreTableOperations;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.SupportsNamespaces;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.exceptions.NamespaceNotEmptyException;
+import org.apache.iceberg.exceptions.NoSuchNamespaceException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.hadoop.HadoopFileIO;
+import org.apache.iceberg.io.FileIO;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.iceberg.CatalogProperties.WAREHOUSE_LOCATION;
+import static org.apache.iceberg.CatalogUtil.dropTableData;
+import static org.apache.iceberg.CatalogUtil.loadFileIO;
+
+/**
+ * An in memory catalog implementation to test Iceberg native mode
+ */
+public class InMemoryCatalog
+        extends BaseMetastoreCatalog
+        implements SupportsNamespaces, Catalog
+{
+    private final ConcurrentHashMap<Namespace, Map<String, String>> namespaces = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<TableIdentifier, String> tables = new ConcurrentHashMap<>();
+
+    private String warehouseLocation;
+    private FileIO fileIO;
+
+    @Override
+    public String name()
+    {
+        return "memory";
+    }
+
+    @Override
+    public void initialize(String name, Map<String, String> properties)
+    {
+        this.warehouseLocation = requireNonNull(properties.get(WAREHOUSE_LOCATION), "warehouseLocation is null");
+        String fileIOImpl = properties.get(CatalogProperties.FILE_IO_IMPL);
```

Comment:
static import CatalogProperties.FILE_IO_IMPL
Presto variables not using abbriviation:
s/fileIOImpl/fileIOImplementation/g